pkg/virt-api/webhooks/validating-webhook/admitters/validate-k8s-utils.go

```diff
@@ -53,6 +55,29 @@ var nodeFieldSelectorValidators = map[string]func(string, bool) []string{
 	metav1.ObjectNameField: ValidateNodeName,
 }
 
+// validateFieldValueNotConformingToDNSLabelRules is function that validate value and checks if given affinities are valid
+func validateFieldValueNotConformingToDNSLabelRules(field *field.Path, value string) (causes []metav1.StatusCause) {
+	if value == "" {
+		return
+	}
+
+	errors := validation.IsDNS1123Label(value)
+	if len(errors) != 0 {
+		causes = appendNewStatusCauseForHostNameNotConformingToDNSLabelRules(field, causes, errors)
+	}
+
+	return causes
+}
+
+func appendNewStatusCauseForHostNameNotConformingToDNSLabelRules(field *field.Path, causes []metav1.StatusCause, errors []string) []metav1.StatusCause {
+	return append(causes, metav1.StatusCause{
+		Type: metav1.CauseTypeFieldValueInvalid,
+		Message: fmt.Sprintf("%s does not conform to the kubernetes DNS_LABEL rules : %s",
+			field.String(), strings.Join(errors, ", ")),
+		Field: field.String(),
+	})
+}
```

Comment:
The name function is still  referring to hostname, while this should be generic too.
Please rename it or drop itand move the content directly inside the caller.
Thanks